### PR TITLE
Added sensor name/id to db/log

### PR DESCRIPTION
**Pull request by nsmfoo**
As requested in #223

### CODE REVIEW

**Comment by katkad** [comment]
test is failing https://travis-ci.org/mushorg/glastopf/builds/73851379

**Comment by nsmfoo** [comment]
Could be an issue with the testing files in the testing directory. The helper.py builds a test config which rightfully lack "sensorid" but it also miss other items.  Maybe @glaslos can shed some light over this issue?

**Comment by glaslos** [comment]
Yes, in https://github.com/mushorg/glastopf/blob/master/glastopf/testing/test_honeypot.py#L50 we generate the config. You want to update here: https://github.com/mushorg/glastopf/blob/master/glastopf/testing/helpers.py#L98
We might want to get rid of that and use the glastopf.conf.dist file instead (having only one place to add new configuration settings).

File: glastopf/glastopf.cfg.dist [hunk 1]
```diff
@@ -99,3 +99,6 @@ banner = Apache/2.0.48
 google_meta =
 #http://www.bing.com/toolbox/webmaster
 bing_meta =
+
+[sensor]
+sensorid = hooli
```
Comment (glaslos) [inline glastopf/glastopf.cfg.dist]: hm, how about setting this to None and change it to a UUID if not set on the first startup?

**Comment by nsmfoo** [comment]
I did think about setting the "sensorid" to the hostname at first, as that would probably be what people set anyway, but decided against it. But a unique UUID, would work.

File: glastopf/glastopf.py [hunk 2]
```diff
@@ -70,8 +70,15 @@ def __init__(self, config="glastopf.cfg", work_dir=os.getcwd()):
             "gid": conf_parser.get("webserver", "gid").encode('latin1'),
             "proxy_enabled": conf_parser.get("webserver", "proxy_enabled").encode('latin1'),
             "banner": conf_parser.get("misc", "banner").encode('latin1'),
+            "sensorid": conf_parser.get("sensor", "sensorid").encode('latin1'),
         }
 
+        if self.options['sensorid'] == "None":
+            self.options['sensorid'] = str(uuid.uuid4())
+            conf_parser.set('sensor', 'sensorid', self.options['sensorid'])
+            with open((os.path.join(self.work_dir,config)), 'wb') as configfile:
```
Comment (glaslos) [inline glastopf/glastopf.py]: Have you seen this: http://www.voidspace.org.uk/python/configobj.html#writing-a-config-file

Comment (glaslos) [inline glastopf/glastopf.py]: Ignore my comment, looks good :+1:

**Comment by glaslos** [comment]
LGTM

File: glastopf/glastopf.py [hunk 3]
```diff
@@ -70,8 +71,15 @@ def __init__(self, config="glastopf.cfg", work_dir=os.getcwd()):
...
+             conf_parser.write(configfile)
```
Comment (glaslos) [inline glastopf/glastopf.py]: Bad indentation. Found 13 spaces, expected 16

**Comment by katkad** [comment]
LGTM

if you could squash commits into one it would be super-awesome :wink: :wink: :sparkles:

**Comment by katkad** [comment]
oh, nice, looks good ;)

**Comment by glaslos** [comment]
@nsmfoo mad git skills!

**Comment by nsmfoo** [comment]
I have @katkad to thank for the mad skills =)

**Comment by katkad** [comment]
working also after squash

@glaslos will you cherry-pick it ?

**Comment by nsmfoo** [comment]
ping?

**Comment by glaslos** [comment]
Merged with 412b39574f516361aeadea6b6fc694d45ae1da29 (sorry for the delay)

**Pull request closed without merging**